frontend/openchat-client/src/openchat.ts

```diff
@@ -1674,6 +1674,13 @@ export class OpenChat extends EventTarget {
     }
 
     removeChat(chatId: string): void {
+        if (this._liveState.myServerChatSummaries[chatId] !== undefined) {
```

Comment:
surely this breaks the mental model. Shouldn't the call to `localChatSummaryUpdates.markRemoved` deal with this? If it doesn't, does it not imply there is something wrong with the code that merges the server with the local data?